src/api/java/com/minecolonies/api/util/WorldUtil.java

```diff
@@ -242,6 +242,12 @@ public static boolean setBlockState(final IWorld world, final BlockPos pos, fina
      */
     public static boolean setBlockState(final IWorld world, final BlockPos pos, final BlockState state, int flags)
     {
+        // To avoid infinite loops or similar we return here just true. We don't want to crash the world or anything.
+        if (pos.getY() < 3)
```

Comment:
why `< 3`? just call world is outside build height (or what's the method name)

Reply:
`< 3` will probably behave inappropriately in 1.17...

Reply:
this really does not belong into the setblockstate function :D. Make sure the callers use the right positions

Reply:
I can do the build hight check. I put it here, so I don't have to check in 50 other places =D

Reply:
yea I did notice that this was for laziness :D

Reply:
but like this only works because we happen to have the override for setblockstate, normally you'd have to make sure proper input values before

Reply:
Yep, isn't that nice =D

Reply:
no its literally a bandaid for calculating bad values :D

Reply:
we already check X/Z boundaries at some place for buildings, should be able to just add Y there aswell

Reply:
Will have to be adjusted in 1.17, I double checked and couldn't find a function returning a min height, just the max height

Reply:
The only reason I bandaid this here is because this crashes, so, this avoids most of the related crashes directly

Reply:
thats like literally the same as surrounding it with a try-catch and leaving the functions trying to use bad positions be, fix those functions instead